apps/server/src/modules/legacy-school/uc/school-system-options.uc.ts

```diff
@@ -0,0 +1,86 @@
+import { ObjectId } from '@mikro-orm/mongodb';
+import { AuthorizationContextBuilder, AuthorizationService } from '@modules/authorization';
+import { System, SystemService } from '@modules/system';
+import { Injectable } from '@nestjs/common';
+import { NotFoundLoggableException } from '@shared/common/loggable-exception';
+import { Permission } from '@shared/domain/interface';
+import { EntityId } from '@shared/domain/types';
+import { AnyProvisioningOptions, SchoolSystemOptions, SchoolSystemOptionsBuilder } from '../domain';
+import { ProvisioningOptionsInterface } from '../interface';
+import { MissingProvisioningStrategyLoggableException } from '../loggable';
+import { SchoolSystemOptionsService } from '../service';
+
+@Injectable()
+export class SchoolSystemOptionsUc {
+	constructor(
+		private readonly authorizationService: AuthorizationService,
+		private readonly systemService: SystemService,
+		private readonly schoolSystemOptionsService: SchoolSystemOptionsService
+	) {}
+
+	public async getProvisioningOptions(
+		userId: EntityId,
+		schoolId: EntityId,
+		systemId: EntityId
+	): Promise<AnyProvisioningOptions> {
+		const schoolSystemOptions: SchoolSystemOptions | null =
+			await this.schoolSystemOptionsService.findBySchoolIdAndSystemId(schoolId, systemId);
+
+		if (!schoolSystemOptions) {
+			throw new NotFoundLoggableException(SchoolSystemOptions.name, { schoolId, systemId });
+		}
+
+		const user = await this.authorizationService.getUserWithPermissions(userId);
+		this.authorizationService.checkPermission(
+			user,
+			schoolSystemOptions,
+			AuthorizationContextBuilder.read([Permission.SCHOOL_SYSTEM_VIEW])
+		);
+
+		return schoolSystemOptions.provisioningOptions;
+	}
+
+	public async setProvisioningOptions(
+		userId: EntityId,
+		schoolId: EntityId,
+		systemId: EntityId,
+		requestedProvisioningOptions: ProvisioningOptionsInterface
+	): Promise<AnyProvisioningOptions> {
+		const system: System | null = await this.systemService.findById(systemId);
+
+		if (!system) {
+			throw new NotFoundLoggableException(System.name, { id: systemId });
+		}
+
+		if (!system.provisioningStrategy) {
+			throw new MissingProvisioningStrategyLoggableException(systemId);
```

Comment:
You are checking the provisioningStrategy in each layer. In my opinion the service would be correct place at the moment because this could be exported and used in other modules.

Or maybe more better at construction of the options. Then it would be near to the domain object and one step forward domain object validation